### DISCUSSION TITLE
ReceiveNative: Send and run code

### DISCUSSION THREAD

**Discussion by Stebalien** [filecoin-project/FIPs#608] [Enhancements - Technical]
Currently, the default way to transfer funds is to invoke method-0. However, this method is guaranteed to not run code.

It would be nice to have a standard way to transfer funds that _does_ run code.

- This would be the _default_ transfer mechanism from user wallets.
- This would be the default when sending from FEVM with no parameters.

This would allow smart smart contracts to _react_ to deposited funds by default.

-----

Bikeshed: I hate the name `ReceiveNative`, but `Receive` and transfer already taken. This method needs a very generic name as it'll likely be implemented on all contracts by default.

**Comment by scotthconner** [comment]
GasReceive?

**Reply by Stebalien** [reply]
Well, this isn't to receive _gas_. It's to receive FIL.

**Comment by Stebalien** [comment]
So, I've discussed this with my team (and others), and the solution is just kind of messy:

1. It would require changing all tools to invoke this method.
2. It would be really easy to accidentally _not_ invoke this method.

We'd _prefer_ to just (optionally) run code on method-0, but then we'd have to provide some way to transfer value without running any user code.

So I'm withdrawing this proposal for now.

**Reply by anorth** [reply]
> we'd have to provide some way to transfer value without running any user code

The read-only flag could get close to achieving this. Code could run, but it couldn't do anything with any persistent side effects (including send the value further on, which I assume is already a restriction).
